errors/grpc.go

```diff
@@ -0,0 +1,21 @@
+package errros
+
+import (
+	"cosmossdk.io/errors"
+	"google.golang.org/grpc/codes"
+)
+
+const moduleName = "grpc"
+
+var (
+	ErrCancelled        = errors.RegisterWithGRPCCode(moduleName, 1, codes.Canceled, "cancelled")
+	ErrUnkown           = errors.RegisterWithGRPCCode(moduleName, 2, codes.Unknown, "unknown")
+	ErrInvalidArgument  = errors.RegisterWithGRPCCode(moduleName, 3, codes.InvalidArgument, "invalid argument")
+	ErrNotFound         = errors.RegisterWithGRPCCode(moduleName, 4, codes.NotFound, "not found")
+	ErrAlreadyExists    = errors.RegisterWithGRPCCode(moduleName, 5, codes.AlreadyExists, "already exists")
+	ErrPermissionDenied = errors.RegisterWithGRPCCode(moduleName, 6, codes.PermissionDenied, "permission denied")
+	ErrInternal         = errors.RegisterWithGRPCCode(moduleName, 7, codes.Internal, "internal")
+	ErrUnavailable      = errors.RegisterWithGRPCCode(moduleName, 8, codes.Unavailable, "unavailable")
+	ErrUnauthenticated  = errors.RegisterWithGRPCCode(moduleName, 9, codes.Unauthenticated, "unauthenticated")
+	ErrUnimplemented    = errors.RegisterWithGRPCCode(moduleName, 10, codes.Unimplemented, "unimplemented")
+)
```

Comment:
Where are these error codes coming from? It looks like the expected not found error should be status code 5 rather than 4 according to this: https://grpc.github.io/grpc/core/md_doc_statuscodes.html. This is also the expected error code that was reported by the user (in the issue summary).

Reply:
Also see https://pkg.go.dev/google.golang.org/grpc/codes. I think we should be using these error codes rather than defining our own... Looks like there is some confusion around the issue per @technicallyty's above request.

Reply:
> Where are these error codes coming from? It looks like the expected not found error should be status code 5 rather than 4 according to this: https://grpc.github.io/grpc/core/md_doc_statuscodes.html. This is also the expected error code that was reported by the user (in the issue summary).

These error codes are just to make sure we don't have duplicate errors in each codespace. Nothing to do with gRPC status codes. For gRPC errors, we are using the third argument (`codes.Canceled`).